api/proto/config/edgeview.proto

```diff
@@ -0,0 +1,56 @@
+// Copyright(c) 2021 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+package org.lfedge.eve.config;
+option go_package  = "github.com/lf-edge/eve/api/go/config";
+option java_package = "org.lfedge.eve.config";
+
+message EdgeViewConfig {
+    // JWT token for signed info, it contains the dispatcher
+    // endpoint IP:Port, device UUID and expiration time
+    string  token = 1;
+    // dispatcher certificate(s) if it's not well-known CA signed
+    repeated bytes disp_cert_pem = 2;
+    // policy for device access through edge-view
+    DevDebugAccessPolicy dev_policy = 3;
+    // policy access for apps through edge-view
+    AppDebugAccessPolicy app_policy = 4;
+}
+
+// Dev debug policy applicable to edge-view
+message DevDebugAccessPolicy {
+    // device side of edge-view access is allowed or not
+    bool allow_dev = 1;
+    // allow edge-view ssh into device or not, ssh key still need to be
+    // installed separately even the edge-view allows it
+    bool ssh = 2;
+    // allow edge-view shell/* cmd or not
+    bool allow_shell = 3;
+    // allow to issue edge-view shell reboot device or not, provided allow_shell is true
+    bool shell_reboot = 4;
```

Comment:
How will you determne all the shell commands which can cause a reboot?
I think there is a long list from reboot, halt, poweroff, shutdown, plus they can be embedded in various shell constructs such as
echo -n "reb" >/tmp/cmd
echo "oot" >>/tmp/cmd
source /tmp/cmd
(that is a very simple one - possibilities are endless).

So I think allow_shell is a first step and if we want more fine grained control it might make sense to add a way to specify which capabilities (https://man7.org/linux/man-pages/man7/capabilities.7.html) the shell should be spawned with and that by default it has none.
Thus to allow a reboot a field in the API (repeated string shell_capabilitiies?) would include "sys_boot" and if the user should be able to look at iptables and manipulate routes etc, then include "net_admin".

FWIW I tried this in EVE-OS by running 
setpriv --inh-caps -net_admin --ambient-caps -net_admin --bounding-set -net_admin /bin/bash
and the new shell can not do network stuff even though it is running as root. There might be easier ways to invoke a shell with a specified set of capabiltiies though.

Reply:
Perhaps we can start with just a "bool priviledged_shell" in the API and consider later adding a list ot capabilties?